pkg/controller/common/license/verifier.go

```diff
@@ -229,6 +229,22 @@ func (l EnterpriseLicense) SignableContentBytes() ([]byte, error) {
 	})
 }
 
+// unescapedJSONMarshal is a custom JSON encoder that turns off Go json's default behaviour of escaping > < and &
+// which is problematic and would lead to failed signature checks as our license signing does not escape those characters.
+func unescapedJSONMarshal(t interface{}) ([]byte, error) {
+	buffer := &bytes.Buffer{}
+	encoder := json.NewEncoder(buffer)
+	encoder.SetEscapeHTML(false)
+	err := encoder.Encode(t)
+	if err != nil {
+		return nil, err
+	}
+	marshaledBytes := buffer.Bytes()
+	// json.Encoder adds an additional newline between objects which we do not want here
+	//as it is not part of the signature, so we are removing the last byte
+	return marshaledBytes[:len(marshaledBytes)-1], err
```

Comment:
I'm pretty convinced the slice manipulation will almost never panic (unless empty content in the license?), however I think it'd still be safer to rely on `bytes.TrimSuffix()` here.

Reply:
Ha I had that initially and though it was overkill. But you are right.